projects/driver_controls/src/pedal_fsm.c

```diff
@@ -0,0 +1,86 @@
+#include "pedal_fsm.h"
+#include "input_event.h"
+#include "event_arbiter.h"
+#include "log.h"
+
+// Pedal FSM state definitions
+
+FSM_DECLARE_STATE(state_brake);
+FSM_DECLARE_STATE(state_coast);
+FSM_DECLARE_STATE(state_driving);
+FSM_DECLARE_STATE(state_cruise_control);
+
+// Pedal FSM transition table definitions
+
+FSM_STATE_TRANSITION(state_brake) {
```

Comment:
I think we should definitely comment all the different conditions that we're handling here :P